samples/bluetooth/alexa_gadget/src/gadgets/CMakeLists.txt

```diff
@@ -0,0 +1,206 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# The Gadgets Profile relies on the nanopb generator tool, which must be
+# available at the configuration stage.
+# This tool is used to compile the alexa .proto files at build time.
+# Output from the .proto compilation is a set of .c and .h files.
+
+# Note: as there doesn't seem to be an easy way of checking the version of the nanopb generator (protoc-gen-nanopb binary)
+# there is the possibility of a version mismatch between the generator and the nanopb source files that is used in the firmware sample.
+# If there is a mismatch, there will be a compilation error in the build step.
+# The easiest way to fix such an error is to download the appropriate nanopb generator .zip and extract it to this sample folder.
+# (https://jpa.kapsi.fi/nanopb/download/)
+
+# Check local folder for protoc and nanopb binaries presence, if not already provided
+if (NOT LOCAL_PROTOC_BIN)
+  file(GLOB_RECURSE LOCAL_PROTOC_BIN "${PROJECT_SOURCE_DIR}/*/generator-bin/protoc")
+  if (NOT LOCAL_PROTOC_BIN)
+      file(GLOB_RECURSE LOCAL_PROTOC_BIN "${PROJECT_SOURCE_DIR}/*/generator-bin/protoc.exe")
+  endif()
+endif()
+
+if (NOT LOCAL_NANOPB_BIN)
+  file(GLOB_RECURSE LOCAL_NANOPB_BIN "${PROJECT_SOURCE_DIR}/*/generator-bin/protoc-gen-nanopb*")
+endif()
```

Comment:
Why is this needed ?
when using `find_program()` below, then supporting a `-D<PROGRAM>` is is already a supported feature.
Like:
```
$ cmake -GNinja -DBOARD=nrf52840dk_nrf52840 -Bbuild samples/bluetooth/alexa_gadget/
...
CMake Error at src/gadgets/CMakeLists.txt:47 (message):
  nanopb required by this project.
...
```
manually specifying `PROTOC_BIN` is handled by `find_program()` instead of `LOCAL_PROTOC_BIN`.
```
$ cmake -GNinja -DBOARD=nrf52840dk_nrf52840 -Bbuild -DPROTOC_BIN=/projects/github/ncs/bin/nanopb-0.4.2-linux-x86/generator-bin/protoc -DNANOPB_BIN=/projects/github/ncs/bin/nanopb-0.4.2-linux-x86/generator-bin/protoc-gen-nanopb samples/bluetooth/alexa_gadget/
...
-- Build files have been written to: /projects/github/ncs/nrf/build
$ cat build/CMakeCache.txt |grep PROTOC_BIN
PROTOC_BIN:FILEPATH=/projects/github/ncs/bin/nanopb-0.4.2-linux-x86/generator-bin/protoc
```

If you want to provide an auto search, then a nicer approach could be to do the search, but add the folder(s) found to `find_program(... PATHS <path>)` (or `HINTS`) and from there allow `find_program` to choose the best candidate. 

Also note, that `file(GLOB_RECURSE ...)` returns a list, so if user download and extract the required zip twice, he could end up with:
```
nanopb-0.4.2-linux-x86
nanopb-0.4.2-linux-x86.1
```
in project dir, causing an obscure error later:
```
$  cmake -GNinja -DBOARD=nrf52840dk_nrf52840 -Bbuild samples/bluetooth/alexa_gadget/
...
-- LOCAL_PROTOC_BIN: /projects/github/ncs/nrf/samples/bluetooth/alexa_gadget/nanopb-0.4.2-linux-x86.1/generator-bin/protoc;/projects/github/ncs/nrf/samples/bluetooth/alexa_gadget/nanopb-0.4.2-linux-x86/generator-bin/protoc
...
$ ninja -Cbuild
[2/234] /bin/sh src/gadgets/CMakeFiles/protobuf_generation-d2ad5ed.sh c6db52e496478d96
FAILED: src/gadgets/CMakeFiles/protobuf_generation generated/alexa/protobuf/accessories.pb.c generated/alexa/protobuf/common.pb.c generated/alexa/protobuf/device.pb.c generated/alexa/protobuf/firmware.pb.c generated/alexa/protobuf/directiveParser.pb.c generated/alexa/protobuf/directiveHeader.pb.c generated/alexa/protobuf/eventParser.pb.c generated/alexa/protobuf/eventHeader.pb.c generated/alexa/protobuf/alexaDiscoveryDiscoverDirective.pb.c generated/alexa/protobuf/alexaDiscoveryDiscoverDirectivePayload.pb.c generated/alexa/protobuf/alexaDiscoveryDiscoverResponseEvent.pb.c generated/alexa/protobuf/alexaDiscoveryDiscoverResponseEventPayload.pb.c generated/alexa/protobuf/notificationsSetIndicatorDirective.pb.c generated/alexa/protobuf/notificationsSetIndicatorDirectivePayload.pb.c generated/alexa/protobuf/notificationsClearIndicatorDirective.pb.c generated/alexa/protobuf/notificationsClearIndicatorDirectivePayload.pb.c generated/alexa/protobuf/alertsSetAlertDirective.pb.c generated/alexa/protobuf/alertsSetAlertDirectivePayload.pb.c generated/alexa/protobuf/alertsDeleteAlertDirective.pb.c generated/alexa/protobuf/alertsDeleteAlertDirectivePayload.pb.c generated/alexa/protobuf/alexaGadgetStateListenerStateUpdateDirective.pb.c generated/alexa/protobuf/alexaGadgetStateListenerStateUpdateDirectivePayload.pb.c generated/alexa/protobuf/alexaGadgetSpeechDataSpeechmarksDirective.pb.c generated/alexa/protobuf/alexaGadgetSpeechDataSpeechmarksDirectivePayload.pb.c generated/alexa/protobuf/alexaGadgetMusicDataTempoDirective.pb.c generated/alexa/protobuf/alexaGadgetMusicDataTempoDirectivePayload.pb.c generated/alexa/protobuf/custom_event.pb.c 
/bin/sh src/gadgets/CMakeFiles/protobuf_generation-d2ad5ed.sh c6db52e496478d96
/projects/github/ncs/nrf/samples/bluetooth/alexa_gadget/nanopb-0.4.2-linux-x86/generator-bin/protoc: File does not reside within any path specified using --proto_path (or -I).  You must specify a --proto_path which encompasses this file.  Note that the proto_path must be an exact prefix of the .proto file names -- protoc is too dumb to figure out when two paths (e.g. absolute and relative) are equivalent (it's harder than you think).

```

I suggest to remove the code. If required to stay, then it need some improvements.
```suggestion

```